hs_access_control/models.py

```diff
@@ -1486,59 +1357,91 @@ def can_delete_resource(self, this_resource):
     # check sharing rights
     ##########################################
 
-    def can_share_resource(self, this_resource, this_privilege):
+    def can_share_resource(self, this_resource, this_privilege, user=None):
```

Comment:
Same answer: check_share_resource raises exception, can_share_resource catches it for convenience.